src/components/FileUpload/FileUpload.types.tsx

```diff
@@ -0,0 +1,46 @@
+import { Accept } from 'react-dropzone'
+
+import { GenericComponentProps } from '../../interfaces/GenericComponentProps'
+import {
+	FilePreviewCommonProps,
+	FileWithPreview
+} from './components/FilePreview'
+import { PasswordModalCommonProps } from './components/PasswordModal'
+
+export interface PasswordModalQueueProps {
+	passwordModalQueue: PasswordModalFile[]
+	setPasswordModalQueue: React.Dispatch<
+		React.SetStateAction<PasswordModalFile[]>
+	>
+}
+
+export interface PasswordModalFile {
+	callback?: (password: string) => void
+	password?: string
+	incorrect?: boolean
+	filename?: string
```

Comment:
I actually thought of that scenario (same named files in two different folders) but our component doesn't support it, regardless of password-protected files. Do you think it's a relevant case that we should support? Uf, but in that case, I would have to take a look at the whole component because for some reason it's not working. When you add the same named file, nothing happens, nothing is added....

Reply:
Yeah, If it doesn't work right know than it's not related to this feature. Sorry, I didn't know that.

I wouldn't handle it right now and I don't know how big your backlog is, but it may be a good idea to add this to some list to make this work in the future.

Reply:
Created a task for that https://linear.app/purple-technology/issue/LAB-30/fileupload-add-support-for-same-named-files